agent/worker.go

```diff
@@ -568,9 +573,9 @@ func (w *worker) Subscribe(ctx context.Context, subscription *api.SubscriptionMe
 		case v := <-ch:
 			task := v.(*api.Task)
 			if match(task) {
-				w.mu.Lock()
+				w.mu.RLock()
```

Comment:
Does the map read not occur after the goroutine launches? I'm not sure how this plays out.